### PR TITLE
added Book:class,repository,service, added configuration for project …

### PR DESCRIPTION
**Pull request by yuriy-kononchuk**
…book-store

### CODE REVIEW

File: src/main/java/com/example/project/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+public class Book {
```
Comment (Rommelua) [inline src/main/java/com/example/project/model/Book.java:16]: add constraints
image

File: src/main/java/com/example/project/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    @Autowired //?? or condtrucror
+    private final BookRepository bookRepository;
+
+    @Autowired //?? or field
+    public BookServiceImpl(BookRepository bookRepository) {
```
Comment (Rommelua) [inline src/main/java/com/example/project/service/impl/BookServiceImpl.java:19]: use constructor

Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/service/impl/BookServiceImpl.java:19]: of course, forgot to clear comments

**Review by olekskov — APPROVED** [review]

File: src/main/java/com/example/project/OnlineBookStoreApplication.java [hunk 3]
```diff
@@ -1,13 +1,29 @@
...
+        return args -> {
+            Book book = new Book();
+            book.setTitle("New Project");
+            book.setAuthor("Yuriy Kononchuk");
+            book.setIsbn("123-456-7890-00-5");
+            book.setPrice(BigDecimal.valueOf(49.99));
+            //bookService.save(book);
+            System.out.println(bookService.findAll());
+        };
+    }*/
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/OnlineBookStoreApplication.java:28]:
```suggestion
```

File: src/main/java/com/example/project/mapper/BookMapper.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.example.project.config.MapperConfig;
+import com.example.project.dto.BookDto;
+import com.example.project.dto.CreateBookRequestDto;
+import com.example.project.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+// Tryed to use automatic generating class BookMapperImpl using MapperConfig
+// but didn't get implementation in /target/generated-sources
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/mapper/BookMapper.java:11]:
```suggestion
```

File: src/main/java/com/example/project/mapper/BookMapperImpl.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        book.setTitle(requestDto.getTitle());
+        book.setAuthor(requestDto.getAuthor());
+        book.setIsbn(requestDto.getIsbn());
+        book.setPrice(requestDto.getPrice());
+        book.setDescription(requestDto.getDescription());
+        book.setCoverImage(requestDto.getCoverImage());
+        return book;
+    }
+}
+
```
Comment (kozhukhovsky) [inline src/main/java/com/example/project/mapper/BookMapperImpl.java:35]:
```suggestion
```

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**